lib/changelogMarkdownParser.js

```diff
@@ -4,7 +4,7 @@ function parseChangelogMarkdown(markdown) {
     const changelog = {};
     let currentVersion;
     let currentLocale;
-    markdown.split('\n').map(line => line.trim()).forEach((line) => {
+    markdown.split('\n').forEach((line) => {
```

Comment:
Why have you removed the trimming here? For instance the new parser will no longer match

```markdown
{SPACE}## 2.0.0
``` 

as a new version header. In my opinion there is no need to change this behaviour here.

Reply:
```
{SPACE}## 2.0.0
```

would be kinda malformed, though

Reply:
> Why have you removed the trimming here? 

Removing the trimming of the changelog lines is essential for the PR because leading spaces belong to the syntax of markdown, especially that is how nested lists work. 

> would be kinda malformed, though

Maybe, but Markdown still converts this to a header. 

We should change the regex to match header lines with leading spaces also.